rasa_nlu/extractors/duckling_extractor.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import unicode_literals
+from __future__ import print_function
+from __future__ import division
+from __future__ import absolute_import
+
+from typing import Text
+
+from rasa_nlu.components import Component
+
+
+DUCKLING_PROCESSING_MODES = ["replace", "append"]
+
+
+class DucklingExtractor(Component):
+    """Adds entity normalization by analyzing found entities and transforming them into regular formats."""
+
+    name = "ner_duckling"
+
+    context_provides = {
+        "process": ["entities"],
+    }
+
+    output_provides = ["entities"]
+
+    def __init__(self, duckling=None):
+        self.duckling = duckling
+
+    @classmethod
+    def cache_key(cls, model_metadata):
+        # type: (Metadata) -> Text
+
+        return cls.name + "-" + model_metadata.language
+
+    def pipeline_init(self, language, duckling_processing_mode):
+        # type: (Text, Text) -> None
+        from duckling import DucklingWrapper
+
+        if duckling_processing_mode not in DUCKLING_PROCESSING_MODES:
+            raise ValueError("Invalid duckling processing mode. Got '{}'. Allowed: {}".format(
+                duckling_processing_mode, ", ".join(DUCKLING_PROCESSING_MODES)))
+
+        # If fine tuning is disabled, we do not need to load the spacy entity model
```

Comment:
I do not understand how fine tuning is related to `self.duckling` being `None` and also not the role the spacy entity model plays in the last lines of this method